doc/Language/grammar_tutorial.pod6

```diff
@@ -567,6 +567,40 @@ Here's the final code:
     }
     =end code
 
+=head2 Add actions directly
+
+Above we see how to associate grammars with action objects and perform
+actions on the match object. However, when we want to deal with the match
+object, that isn't the only way. See the example below:
+
+    =begin code
+    grammar G {
+      rule TOP { <function-define> }
+      rule function-define {
+        'sub' <identifier> {
+          say "func " ~ $<identifier>.made;
+          make $<identifier>.made;
+        } '(' <parameter> ')' '{' '}' {
+          say "end " ~ $/.made;
+        }
+      }
+      token identifier { \w+ { make ~$/; } }
+      token parameter { \w+ { say "param " ~ $/; } }
+    }
+
+    G.parse('sub f ( a ) { }');
+    # OUTPUT: «func f␤param a␤end f␤»
+    =end code
+
+This example is a reduced portion of a parser. Let's focus more on the feature
+it shows.
+
+First, we can add actions inside the grammar ifself, and such actions are
+performed once the match arrives at them. Note that action object's method will
```

Comment:
Technically it's not the match that arrives at the actions, but rather the control flow of the regex. (The match is related to the position in the string, not directly to the position within the regex).

Reply:
That is what I want to express